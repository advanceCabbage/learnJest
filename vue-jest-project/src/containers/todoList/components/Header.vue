<template>
  <div class="header">
    <div class="header-content">
      <input
        type="text"
        data-test="input"
        :value="inputValue"
        @keyup.enter="addTodoItem"
        @input="(e) => changeInputValue(e.target.value)"
        placeholder="add todo"
      />
    </div>
  </div>
</template>

<script>
import { mapState, mapMutations } from "vuex";
export default {
  name: "Header",
  props: {},
  computed: {
    ...mapState({
      inputValue: (state) => state.inputValue,
    }),
  },
  methods: {
    addTodoItem() {
      if (this.inputValue) {
        this.$emit("add", this.inputValue);
        this.changeInputValue("");
      }
    },
    ...mapMutations({ changeInputValue: "changeInputValue" }),
  },
};
</script>

<style scoped lang="stylus">
.header{
  line-height: 60px;
  background-clip: #666;
}
.header-content{
  width 60px;
  height: 60px
}
</style>
