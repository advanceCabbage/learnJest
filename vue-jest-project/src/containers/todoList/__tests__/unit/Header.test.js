import { shallowMount } from "@vue/test-utils";
import Header from "@/containers/todoList/components/Header.vue";
import { findTestWrapper } from "../../../../utils/testUtils";

describe("Header组件", () => {
  it("存在input框", () => {
    const wrapper = shallowMount(Header);
    const input = findTestWrapper(wrapper, "input");
    expect(input.exists()).toBe(true); // input 元素是否存在
  });

  it("input框 初始内容为空", () => {
    const wrapper = shallowMount(Header);
    const inputValue = wrapper.vm.$data.inputValue;
    expect(inputValue).toBe("");
  });

  it("input发生变化 数据跟随变化", () => {
    const wrapper = shallowMount(Header);
    const input = findTestWrapper(wrapper, "input");
    input.setValue("dell lee");
    const inputValue = wrapper.vm.$data.inputValue;
    expect(inputValue).toBe("dell lee");
  });

  it("input 输入回车，无内容时，无反应", () => {
    const wrapper = shallowMount(Header);
    const input = findTestWrapper(wrapper, "input");
    input.setValue("");
    input.trigger("keyup.enter");
    expect(wrapper.emitted().add).toBeFalsy(); // 回车为空时 不应该向外面触发一个add事件
  });

  it("input 输入回车，有内容时，向外触发事件,同时清空inputValue", () => {
    const wrapper = shallowMount(Header);
    const input = findTestWrapper(wrapper, "input");
    input.setValue("dell lee");
    input.trigger("keyup.enter");
    expect(wrapper.emitted().add).toBeTruthy();
    const inputValue = wrapper.vm.$data.inputValue;
    expect(inputValue).toBe("");
  });

  it("header 样式发生改变，做提示", () => {
    const wrapper = shallowMount(Header);
    expect(wrapper).toMatchSnapshot();
  });
});
