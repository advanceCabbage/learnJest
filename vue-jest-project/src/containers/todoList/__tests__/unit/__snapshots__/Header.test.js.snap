// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header组件 header 样式发生改变，做提示 1`] = `
<div class="header">
  <div class="header-content"><input type="text" data-test="input" placeholder="add todo"></div>
</div>
`;
